docs/src/donu.md

```diff
@@ -596,3 +597,146 @@ Example:
   }
 }
 ```
+
+### `list-datasets` List available datasets
+
+**Request:**
+
+| Field            | Type                     | Description                                                         |
+|------------------|--------------------------|---------------------------------------------------------------------|
+| command          | string                   | Command - for this operation it will be the string `"list-datasets"`|
+
+Example:
+
+```JSON
+{
+  "command": "list-datasets"
+}
+```
+
+**Response:**
+
+A list of dataset descriptions.
+
+```JSON
+{
+  "status": "success",
+  "result": [
+    {
+      "source": {
+        "model": "example.json"
```

Comment:
I don't love that this calls a dataset a `"model"` - maybe we'd prefer to JSONify a DataSource to hold a `"file"` instead, as we did a while ago? This quibble may not rise to the level of justifying a breaking change.

Reply:
It agree sucks but i don't want to freak out uncharted - if you think it's better we can use "`file"` but i think the right thing is probably to change `"model"` to `"storage"` or something

Reply:
Alright, I'm happy enough to keep it as-is for the time being.